test/functional/apps/console/_xjson.ts

```diff
@@ -39,11 +88,49 @@ export default ({ getService, getPageObjects }: FtrProviderContext) => {
       });
     });
 
-    describe('with invalid syntax', () => {
-      it('should trigger validation errors', async () => {
-        await PageObjects.console.enterRequest('\nGET test/doc/1 \n{\n "foo": \'\'');
-        expect(await PageObjects.console.hasInvalidSyntax()).to.be(true);
-        expect(await PageObjects.console.hasErrorMarker()).to.be(true);
+    describe('inline comments', () => {
+      it('should be correctly syntax highlighted', async () => {
+        const commentTokenColor = '#41755c';
+        await PageObjects.console.enterRequest('\n GET _search // inline comment');
+        const commentTokenColorRGB = await PageObjects.console.getTokenColor('ace_comment');
+        // getTokenColor returns rgb value of css color property, we need to convert rgb to hex to compare it to the actual value
+        expect(rgbToHex(commentTokenColorRGB)).to.eql(commentTokenColor);
+      });
+
+      it('should allow inline comments in request url row', async () => {
+        await PageObjects.console.enterRequest('\n GET _search // inline comment');
+        expect(await PageObjects.console.hasErrorMarker()).to.be(false);
+        await PageObjects.console.clickPlay();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        expect(await PageObjects.console.getResponseStatus()).to.eql(200);
+      });
+
+      it('should allow inline comments in request body', async () => {
+        await PageObjects.console.enterRequest(
+          '\n GET _search \n{\n "query": {\n "match_all": {} // inline comment'
+        );
+        expect(await PageObjects.console.hasErrorMarker()).to.be(false);
+        await PageObjects.console.clickPlay();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        expect(await PageObjects.console.getResponseStatus()).to.eql(200);
+      });
+    });
+
+    describe('with a request using a deprecated feature', () => {
+      it('should print a warning into the response pane above the JSON', async () => {
+        await PageObjects.console.enterRequest('\nGET .kibana');
+        await PageObjects.console.clickPlay();
+        await PageObjects.header.waitUntilLoadingHasFinished();
```

Comment:
nit: does it make sense to move the following 3 lines into separate method and use it across all tests?  

```
await PageObjects.console.enterRequest('\nGET .kibana');
await PageObjects.console.clickPlay();
await PageObjects.header.waitUntilLoadingHasFinished();
```
something like: 

```
const enterRequest = async (body: string) =>  {
      await PageObjects.console.enterRequest('\nGET .kibana');
      await PageObjects.console.clickPlay();
      await PageObjects.header.waitUntilLoadingHasFinished();
} 
....
it('should print a warning into the response pane above the JSON', async () => {
    await enterRequest('\nGET .kibana');
    expect(await PageObjects.console.responseHasDeprecationWarning()).to.be(true);
});
```